src/puya/mir/models.py

```diff
@@ -53,6 +53,30 @@ def __str__(self) -> str:
         return f"byte {format_bytes(self.value, self.encoding)}"
 
 
+@attrs.frozen(eq=False)
+class PushDeployVar(BaseOp):
+    name: str
+    atype: AVMType
+
+    def accept(self, visitor: MIRVisitor[_T]) -> _T:
+        return visitor.visit_push_deploy_var(self)
+
+    @property
+    def op_code(self) -> str:
+        match self.atype:
+            case AVMType.bytes:
+                return "pushbytes"
+            case AVMType.uint64:
+                return "pushint"
+            case _:
+                raise InternalError(
```

Comment:
Maybe move this validation to an `attrs` validator? or even, since it's fixed, make this an `attrs` `.default` method and do the validation at the same time as op code selection, but it'll be done on construction at least.